framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2738,6 +2738,20 @@ private AnnotatedDeclaredType getFunctionalInterfaceType(Tree lambdaTree,
         }
     }
 
+    private AnnotatedTypeMirror unwrapVarargs(List<AnnotatedTypeMirror> parameterTypes, int index) {
+        final int lastIndex = parameterTypes.size() - 1;
+        final AnnotatedTypeMirror lastType = parameterTypes.get(lastIndex);
+        final boolean parameterBeforeVarargs = index < lastIndex;
+        if (!parameterBeforeVarargs && lastType instanceof AnnotatedArrayType) {
+            final AnnotatedArrayType arrayType = (AnnotatedArrayType) lastType;
+            final Type.ArrayType underlyingType = (Type.ArrayType) arrayType.getUnderlyingType();
```

Comment:
I take that back.  Only the executable type has the isVarArgs method.  Should we add one for AnnotatedArrayType?  It seems like it might be a useful method.